pkg/tar/tar_test.go

```diff
@@ -411,22 +400,10 @@ func TestExtractTarStreamTimeout(t *testing.T) {
 		t.Fatalf("Cannot create temp directory: %v", err)
 	}
 	defer os.RemoveAll(destDir)
-	wg := sync.WaitGroup{}
-	wg.Add(2)
 	th := New()
 	th.(*stiTar).timeout = 10 * time.Millisecond
-	go func() {
-		defer wg.Done()
-		time.Sleep(20 * time.Millisecond)
-		writer.Close()
-	}()
-	extractError := make(chan error, 1)
-	go func() {
-		defer wg.Done()
-		extractError <- th.ExtractTarStream(destDir, reader)
-	}()
-	wg.Wait()
-	err = <-extractError
+	time.AfterFunc(20*time.Millisecond, func() { writer.Close() })
```

Comment:
Because the test purpose is to prove that the set timeout of 10ms is functional.